utils/backoff.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/cenkalti/backoff/v4"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// WithMaxRetriesLog runs the operation until it succeeds or max retries has been reached.
+// It uses exponential back off.
+// It optionally logs information if logger is set.
+func WithMaxRetriesLog(
+	operation backoff.Operation,
+	max uint64,
+	logger logging.Logger,
+	msg string,
+	fields ...zapcore.Field,
+) error {
+	attempt := uint(1)
+	expBackOff := backoff.WithMaxRetries(backoff.NewExponentialBackOff(), max)
```

Comment:
Let's use `backoff.WithMaxElapsedTime` instead. At present, we choose the number of retries and the delay between each to resolve to a set elapsed time before we emit an error.

Reply:
If I understand well, you want to use `backoff.WithMaxElapsedTime` instead of `backoff.WithMaxRetries`.

Instead of giving a `maxRetry` to the utility function, I will give a `maxElapsedTime` derived from the existing values (`numberOfRetries * delayBetweenEach`)